schemas/cloud.sql

```diff
@@ -119,7 +119,7 @@ BEGIN
         END IF;
     END IF;
 
-    INSERT INTO CloudRecords(RecordCreateTime, VMUUID, SiteID, CloudComputeServiceID, MachineName,
+    REPLACE INTO CloudRecords(RecordCreateTime, VMUUID, SiteID, CloudComputeServiceID, MachineName,
```

Comment:
What is the benefit of using REPLACE here?

Reply:
The benefit is we don't have to think about how the "upsert" is handled, as we currently do with the `INSERT... ON DUPLICATE KEY UPDATE`, `REPLACE` just keeps the last one.

We could achieve the same with `INSERT... ON DUPLICATE KEY UPDATE`, but it wouldn't reduce the complexity of the `ON DUPLICATE KEY UPDATE` very much.

Reply:
Ah, so we were limited to using `INSERT... ON DUPLICATE KEY UPDATE` before as we needed to do fiddly things with MeasurementTime, right? But now we're not limited to that so `REPLACE` is the simpler option?

Reply:
👍